crates/solvers/src/domain/eth/mod.rs

```diff
@@ -57,6 +73,18 @@ impl std::ops::Add for Gas {
     }
 }
 
+impl std::ops::Add<SignedGas> for Gas {
+    type Output = Self;
+
+    fn add(self, rhs: SignedGas) -> Self::Output {
```

Comment:
We should also consider the sign of the `self`, what if `self` is also negative?